### PR TITLE
Serving SRV targets and getting all nodes in single request

### PR DESCRIPTION
**Pull request by adnull**
As it stated in https://github.com/percona/mongodb_exporter/issues/773 there is no chance to monitor cluster nodes with the only mongodb+srv url specified. 
According to the specification, direct connection couldn't be used with a cluster, specified as a SRV domain

This update adds 
- --split-cluster option, which:
  - extracts mongodb+srv domain into cluster connection string containing nodes from DNS server
  - splits all cluster connection string into separate targets that can be queried by "target" parameter of multitarget endpoint
- overall target endpoint "/scrapeall" which outputs all the targets metrics with a "instance" label containing nodename as a host:port pair (or only host if no port specified)

Combining all that features we can specify cluster url as mongodb+srv domain and monitor all the nodes metrics and states in one request by querying /scrapeall endpoint

### CODE REVIEW

**Review by JiriCtvrtka — APPROVED** [review]

File: exporter/server.go [hunk 1]
```diff
@@ -97,6 +104,68 @@ func multiTargetHandler(serverMap ServerMap) http.HandlerFunc {
 	}
 }
 
+// OverallTargetsHandler is a handler to scrape all the targets in one request.
+// Adds instance label to each metric.
+func OverallTargetsHandler(exporters []*Exporter, logger *logrus.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		seconds, err := strconv.Atoi(r.Header.Get("X-Prometheus-Scrape-Timeout-Seconds"))
+		// To support also older ones vmagents.
```
Comment (idoqo) [inline exporter/server.go:112]:
```suggestion
		// To support older vmagents.
```

File: exporter/server.go [hunk 2]
```diff
@@ -97,6 +104,68 @@ func multiTargetHandler(serverMap ServerMap) http.HandlerFunc {
...
+		if err != nil {
+			seconds = 10
```
Comment (idoqo) [inline exporter/server.go:114]: Helpful to log this? So folks don't scratch their heads when their set timeout isn't working.

File: exporter/server.go [hunk 3]
```diff
@@ -97,6 +104,68 @@ func multiTargetHandler(serverMap ServerMap) http.HandlerFunc {
...
+		}
+
+		var gatherers prometheus.Gatherers
+		gatherers = append(gatherers, prometheus.DefaultGatherer)
+
+		filters := r.URL.Query()["collect[]"]
+
+		for _, e := range exporters {
+			ctx, cancel := context.WithTimeout(r.Context(), time.Duration(seconds-e.opts.TimeoutOffset)*time.Second)
+			defer cancel()
```
Comment (idoqo) [inline exporter/server.go:124]: These defer (and the one on L135) won't run when each loop ends though, is that intentional?

Comment (adnull) [reply exporter/server.go:124]: As I understand the logic we create gatherers in the loop, but real metrics collecting is being done by promhttp.HandlerFor which calls collector.Collect and it is located outside the loop, so we free the resources after the all work done.

File: exporter/server.go (hunk 2 above)
Comment (adnull) [reply exporter/server.go:114]: logged that as a debug message

**Review by idoqo — APPROVED** [review]

File: REFERENCE.md [hunk 4]
```diff
@@ -1,34 +1,34 @@
...
+|--collector.currentopmetrics|Enable collecting metrics from currentop admin command|
+|--collector.indexstats|Enable collecting metrics from $indexStats|
+|--collector.collstats|Enable collecting metrics from $collStats|
+|--collect-all|Enable all collectors. Same as specifying all --collector.\<name\>|
+|--collector.collstats-limit=0|Disable collstats, dbstats, topmetrics and indexstats collector if there are more than \<n\> collections. 0=No limit|
+|--collector.profile-time-ts=30|Set time for scrape slow queries| This interval must be synchronized with the Prometheus scrape interval|
+|--collector.profile|Enable collecting metrics from profile|
+|--collector.shards|Enable collecting metrics related to Mongo shards|
+|--metrics.overridedescendingindex| Enable descending index name override to replace -1 with _DESC ||
+|--version|Show version and exit|
```
Comment (BupycHuk) [inline REFERENCE.md:34]: `--collector.pbm` is lost

**Review by BupycHuk — APPROVED** [review]

File: REFERENCE.md (hunk 4 above)
Comment (adnull) [reply REFERENCE.md]: damn, I checked the list after merge but missed it. fixed.

File: exporter/exporter.go [hunk 5]
```diff
@@ -386,6 +354,42 @@ func (e *Exporter) Handler() http.Handler {
...
+		case "indexstats":
+			requestOpts.EnableIndexStats = true
+		case "collstats":
+			requestOpts.EnableCollStats = true
+		case "profile":
+			requestOpts.EnableProfile = true
+		case "shards":
+			requestOpts.EnableShards = true
+		case "pbm":
+			requestOpts.EnablePBMMetrics = true
```
Comment (idoqo) [inline exporter/exporter.go:386]: seems FCV was missed during the merge:
```suggestion
		case "pbm":
			requestOpts.EnablePBMMetrics = true
		case "fcv":
			requestOpts.EnableFCV = true
		
```

**Review by idoqo — CHANGES_REQUESTED** [review]
Looks good, we should also enable fcv collector.

File: exporter/exporter.go (hunk 5 above)
Comment (adnull) [reply exporter/exporter.go:388]: Omg, how did that happen.. fixed

**Review by idoqo — APPROVED** [review]

**Pull request merged**